src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs

```diff
@@ -180,7 +179,13 @@ private static SyntaxNode GetEnclosingCodeElementNode(Document document, SyntaxT
 
                     await Task.WhenAll(referenceTasks).ConfigureAwait(false);
 
-                    return referenceTasks.Select(task => task.Result);
+                    var result = ArrayBuilder<ReferenceLocationDescriptor>.GetInstance();
+                    foreach (var task in referenceTasks)
+                    {
+                        result.Add(await task.ConfigureAwait(false));
```

Comment:
the correct thing to do is actually 'await' the task.  This ensures that if it Canceled (or threw some other exception) that that will be processed correctly.